### PR TITLE
Add close new window

### LINKED ISSUE #39: Update step to close new window

### PR DESCRIPTION
**Pull request by pradip3271**
Fixes #39 
Proposed changes in this PR
- Remove Then I should see url pag step def.
- Add close new tab in close child window method

### CODE REVIEW

File: src/Context/CommonContext.php [hunk 1]
```diff
@@ -391,18 +391,19 @@ public function iShouldNotSeeElementByCss($selector): void
      */
     public function thenIShouldSeePage(string $url): void
     {
+        $current_url = $this->getSession()->getCurrentUrl();
         $windowNames = $this->getSession()->getWindowNames();
         if (count($windowNames) > 1) {
             $this->switchToNewTab($windowNames);
+            $current_url = $this->getSession()->getCurrentUrl();
```
Comment (VenuBotla) [inline src/Context/CommonContext.php]: Why are you overriding $current_url in this step.

File: src/Context/CommonContext.php [hunk 2]
```diff
@@ -391,18 +391,19 @@ public function iShouldNotSeeElementByCss($selector): void
      */
     public function thenIShouldSeePage(string $url): void
     {
+        $current_url = $this->getSession()->getCurrentUrl();
```
Comment (VenuBotla) [inline src/Context/CommonContext.php]: What is the use of this step?

File: src/Context/CommonContext.php (hunk 1 above)
Comment (pradip3271) [reply src/Context/CommonContext.php]: if it more than 1 window it take the new window url, otherwise it take the main window url.

Comment (VenuBotla) [reply src/Context/CommonContext.php]: What is the use case for this method?
We already have a method to verify the page url. If you are looking to verify the url of the new window then you should not have if condition.

File: src/Context/CommonContext.php [hunk 3]
```diff
@@ -391,18 +391,19 @@ public function iShouldNotSeeElementByCss($selector): void
...
         $windowNames = $this->getSession()->getWindowNames();
         if (count($windowNames) > 1) {
             $this->switchToNewTab($windowNames);
+            $current_url = $this->getSession()->getCurrentUrl();
+            $this->getSession()->executeScript('window.close()');
```
Comment (ayanozturk) [inline src/Context/CommonContext.php]: This is introducing a new behaviour and closing the page/tab, which it was not doing before and it is not clear on this methods name or gherkin language definition. If it is a new behaviour, it should be a separate method or it should be made clear on definition of this content implementation.

Also: according to PSR standards, please use `$camelCase` variable names.

File: src/Context/CommonContext.php [hunk 4]
```diff
@@ -384,41 +429,29 @@ public function iShouldNotSeeElementByCss($selector): void
         $this->assertSession()->elementNotExists('css', $selector);
     }
 
-    /**
-     * @Then I should see :url page url
-     * @param string $url
-     * @throws \Exception
-     */
-    public function thenIShouldSeePage(string $url): void
```
Comment (ayanozturk) [inline src/Context/CommonContext.php]: Are we sure that it is safe to remove this context method? This is a library and there might be applications testing functionality depending on this code.

Standard practice to remove code from libraries is: marking the methods as `@deprecated` and release a new major version with all deprecated methods removed and BC upgrade path and changes documented.

Comment (pradip3271) [reply src/Context/CommonContext.php]: @ayanozturk I have added context method, Venu said its duplicate and few tests are using those method in SR and CR projects. So I have to change those once it merged.

Comment (adamclark-dev) [reply src/Context/CommonContext.php]: Agree with @ayanozturk here, we should mark it as deprecated and remove in the next major version.

File: src/Context/CommonContext.php [hunk 5]
```diff
@@ -531,7 +534,9 @@ public function iCloseTheChildWindow()
     {
         $windowNames = $this->getSession()->getWindowNames();
         if (count($windowNames) > 1) {
-            $this->getSession()->stop($windowNames[1]);
+            $this->getSession()->executeScript('window.close()');
+            //switch back to main window
+            $this->getSession()->switchToWindow();
```
Comment (ayanozturk) [inline src/Context/CommonContext.php]: - Revert changes in this method and mark is as `@deprecated`
- Add a new method with new step definition and new behaviour

File: src/Context/CommonContext.php [hunk 6]
```diff
@@ -400,24 +400,27 @@ public function iShouldNotSeeElementByCss($selector): void
...
+            $current_url = $this->getSession()->getCurrentUrl();
+            $this->getSession()->executeScript('window.close()');
+            //switch back to main window
+            $this->getSession()->switchToWindow();
         }
 
-        $current_url = $this->getSession()->getCurrentUrl();
         if (!strpos($current_url, $url)) {
-            throw new \RuntimeException("Can not find url $url");
+            throw new \Exception("Can not find url $url");
```
Comment (ayanozturk) [inline src/Context/CommonContext.php]: revert changes

File: src/Context/CommonContext.php [hunk 7]
```diff
@@ -400,24 +400,27 @@ public function iShouldNotSeeElementByCss($selector): void
...
     public function thenIShouldSeePage(string $url): void
     {
+        $current_url = $this->getSession()->getCurrentUrl();
         $windowNames = $this->getSession()->getWindowNames();
         if (count($windowNames) > 1) {
             $this->switchToNewTab($windowNames);
+            $current_url = $this->getSession()->getCurrentUrl();
+            $this->getSession()->executeScript('window.close()');
+            //switch back to main window
+            $this->getSession()->switchToWindow();
```
Comment (ayanozturk) [inline src/Context/CommonContext.php]: revert changes

**Pull request merged**